src/Bicep.LangServer/Handlers/BicepDeploymentScopeRequestHandler.cs

```diff
@@ -134,6 +134,8 @@ private Compilation GetCompilation(DocumentUri documentUri)
         {
             var fileUri = documentUri.ToUri();
 
+            // Bicep file could contain load functions like loadTextContent(..). We'll refresh compilation to detect changes in files referenced in load functions.
+            compilationManager.RefreshCompilation(fileUri);
```

Comment:
@bhsubra the following code appears both in build and deploy commands:
```cs
            CompilationContext? context = compilationManager.GetCompilation(fileUri);
            Compilation compilation;

            if (context is null)
            {
                SourceFileGrouping sourceFileGrouping = SourceFileGroupingBuilder.Build(this.fileResolver, this.moduleDispatcher, new Workspace(), fileUri);
                compilation = new Compilation(featureProviderFactory, namespaceProvider, sourceFileGrouping, configurationManager, apiVersionProviderFactory, bicepAnalyzer);
            }
            else
            {
                compilation = context.Compilation;
            }
```
a) should it be extracted?
b) when would context be null?  
c) should we just be created a new compilation each time?

Reply:
Extracted the above mentioned code here:
https://github.com/Azure/bicep/commit/44fef02d7a149af50a079ad9f5be86fbb022f14a

CompilationContext would be null if the file is not open. E.g. User right clicks on a file from explorer context menu, without opening the file and invokes build/deploy. We should only create a new compilation if we detect the compilation to be null. Let me know if you have any concerns here.

Thanks!

Reply:
Thanks!